x/sequencer/keeper/proposer.go

```diff
@@ -37,90 +37,38 @@ func (k Keeper) RollappPotentialProposers(ctx sdk.Context, rollappId string) []t
 	return append(seqs, k.SentinelSequencer(ctx))
 }
 
-// ChooseProposer will assign a proposer to the rollapp. It won't replace the incumbent proposer
-// if they are not sentinel. Otherwise it will prioritise a non sentinel successor. Finally, it
-// choose one based on an algorithm.
-// The result can be the sentinel sequencer.
-func (k Keeper) ChooseProposer(ctx sdk.Context, rollapp string) error {
+// RecoverFromHalt will assign a new proposer to the rollapp.
+// It will choose a new proposer from the list of potential proposers.
+// The rollapp must be halted and with potential proposer available.
+func (k Keeper) RecoverFromHalt(ctx sdk.Context, rollapp string) error {
```

Comment:
I suggest `recoverFromHalt` -> `recoverFromSentinel` - halt is out of context in the L1 (i.e Its a potential implication on the L2 which I think can be confusing for the reader ).
also it may be thought of halt can thought of as halt of the L1.
also it's always being called after we check sequencer is not sentinel.
so I think overall it's much clearer.